apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -176,87 +177,101 @@ oracle_ttl_t() ->
 sign_t() ->
     {tuple, [word, word]}.
 
-oracle_call_register(_Value, Data, State) ->
+oracle_call_register(_Value, Data, #chain{api = API, state = State} = Chain) ->
     ArgumentTypes = [word, sign_t(), word, oracle_ttl_t(), typerep, typerep],
-    [Acct, Sign0, QFee, TTL, QType, RType] = get_args(ArgumentTypes, Data),
-    case chain_ttl_delta(TTL, State) of
+    [Acct, Sign0, QFee, TTL, QFormat, RFormat] = get_args(ArgumentTypes, Data),
+    %% The aeo_register_tx expects the formats to be binary, althought, atoms are provided.
+    %% This should be solved in a different ticket.
+    case chain_ttl_delta(TTL, Chain) of
         {error, _} = Err -> Err;
         {ok, DeltaTTL = {delta, _}} ->
+            {ok, Tx} = API:oracle_register_tx(<<Acct:256>>, QFee, TTL, QFormat, RFormat, State),
             Callback =
-                fun(API, ChainState) ->
-                    case API:oracle_register(<<Acct:256>>, to_sign(Sign0), QFee, TTL, QType, RType, ChainState) of
+                fun(ChainAPI, ChainState) ->
+                    case ChainAPI:oracle_register(Tx, to_sign(Sign0), ChainState) of
                         {ok, <<OKey:256>>, ChainState1} -> {ok, OKey, ChainState1};
                         {error, _} = Err                -> Err
                     end
                 end,
-            DynGas = state_gas(oracle_registration, DeltaTTL),
-            ?TEST_LOG("~s computed gas cost ~p from relative TTL ~p", [?FUNCTION_NAME, DynGas, DeltaTTL]),
-            {ok, DynGas, fun() -> call_chain(Callback, State) end}
+            SizeGas = size_gas(Tx),
+            StateGas = state_gas(oracle_registration, DeltaTTL),
+            DynGas = SizeGas + StateGas,
```

Comment:
Doc update required to https://github.com/aeternity/protocol/blame/3bc313196b6e0b0319fa892e17d5bb9327c10395/contracts/aevm.md#L35-L42 (including AENS also).

Reply:
Yes, I'm working on it.